test/HTML5/Parser/DOMTreeBuilderTest.php

```diff
@@ -457,14 +457,17 @@ public function testText()
         $data = $wrapper->childNodes->item(0);
         $this->assertEquals(XML_TEXT_NODE, $data->nodeType);
         $this->assertEquals('test', $data->data);
+    }
 
+    public function testTextBeforeHeadNotAllowed()
+    {
         // The DomTreeBuilder has special handling for text when in before head mode.
-        $html = '<!DOCTYPE html><html>
-    Foo<head></head><body></body></html>';
+        $html = '<!DOCTYPE html><html>Foo<head></head><body></body></html>';
         $doc = $this->parse($html);
-        $this->assertEquals('Line 0, Col 0: Unexpected text. Ignoring: Foo', $this->errors[0]);
+        $this->assertEquals('Line 0, Col 0: Unexpected head tag outside of head context.', $this->errors[0]);
         $headElement = $doc->documentElement->firstChild;
         $this->assertEquals('head', $headElement->tagName);
+        $this->assertXmlStringEqualsXmlString($doc, '<html xmlns="http://www.w3.org/1999/xhtml"><head/><body>Foo<head/><body/></body></html>');
```

Comment:
This does not seem to be correct. `body` in `body` should lead to a parsing error according to https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-inbody

Reply:
Yeah that's what I wrote in https://github.com/Masterminds/html5-php/pull/183#issuecomment-632269178

Reply:
it did lead to a parsing error, that is why `$this->errors[0]` has the error in it.